File: Backend/frontend/lib/features/events/data/event_color.dart
class EventColor {
  final String id;
  final String eventId;
  final String colorHex;
  final int sortOrder;

  EventColor({
    required this.id,
    required this.eventId,
    required this.colorHex,
    this.sortOrder = 0,
  });

  factory EventColor.fromJson(Map<String, dynamic> json) {
    return EventColor(
      id: json['id'],
      eventId: json['event_id'],
      colorHex: json['color_hex'],
      sortOrder: json['sort_order'] ?? 0,
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'event_id': eventId,
        'color_hex': colorHex,
        'sort_order': sortOrder,
      };
}

File: Backend/internal/store/models/event_color.go
package models

import (
	"github.com/google/uuid"
)

type EventColor struct {
	ID        uuid.UUID `json:"id"`
	EventID   uuid.UUID `json:"event_id"`
	ColorHex  string    `json:"color_hex"`
	SortOrder int       `json:"sort_order"`
}